include/soci/use-type.h

```diff
@@ -72,6 +72,21 @@ class SOCI_DECL standard_use_type : public use_type_base
         //convert_to_base();
     }
 
+    // Const indicator constructor
+    standard_use_type(void* data, exchange_type type, const indicator& ind,
+        bool readOnly, std::string const& name = std::string())
+        : data_(data)
+        , type_(type)
+        // Const cast will allow the code to compile and run but any modification to ind_ in readonly mode will result in undefined behaviour
```

Comment:
I agree. The other alternative is to add another member( see conversion_use_type.h where 2 indicators are used)

The cause of this complication is the in/out support in use scenario therefore ```readOnly``` flag was added and that complicated things.